plugins/obs-webrtc/whip-output.cpp

```diff
@@ -5,16 +5,19 @@ const char signaling_media_id_valid_char[] = "0123456789"
 					     "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
 					     "abcdefghijklmnopqrstuvwxyz";
 
-const uint32_t audio_ssrc = 5002;
+const uint32_t audio_ssrc = generate_random_u32();
 const char *audio_mid = "0";
 const uint32_t audio_clockrate = 48000;
 const uint8_t audio_payload_type = 111;
 
-const uint32_t video_ssrc = 5000;
+const uint32_t video_ssrc = generate_random_u32();
 const char *video_mid = "1";
 const uint32_t video_clockrate = 90000;
 const uint8_t video_payload_type = 96;
 
+// user-agent to use for requests constructed at Init()
+std::string user_agent;
```

Comment:
Do we need to have this variable in the global scope? It looks this variable is updated from `Init`, which is called from `StartThread`. Is this thread safe?

Reply:
If I understand correctly, `StartThread` calls `Init` at first so that the variable `user_agent` is updated every time the WHIP output started.

Reply:
@norihiro true, what do you suggest as far as making it constant, or am I misunderstanding?

Reply:
If I were you, I would put `user_agent` as a member variable of the class `WHIPOutput` or construct the user agent everytime the user agent is required because it is required only 3 times for each pair of start-stop.
Thinking 3rd party plugins like obs-multi-rtmp and the possibility of more features in obs, in my opinion, the output should be designed to be thread-safe.